src/main/java/ga/rugal/food/springmvc/controller/MenuAction.java

```diff
@@ -38,26 +39,34 @@
     @Autowired
     private ServletContext context;
 
-    private final Random random = new Random();
-
     @Autowired
     private MenuService menuService;
+    
+    @Autowired
+    private RestaurantService restaurantService;
 
+    /**
+     * Get menu information through URL /menu
+     * This method is to get a random menu from a menu list associated with specific restaurant
+     * 
+     * @return Give successful message and menu data in JSON format if the menu exist, 
+     * on the contrary, return failed message if menu can't be found
+     */
     @ResponseBody
     @RequestMapping(method = RequestMethod.GET)
-    public Message randomMenu()
-    {
-        int total = menuService.countTotal();
-        LOG.debug(CommonLogContent.MENU_NUMBER, total);
-        if (0 == total)
-        {
+    public Message menuByRestaurant() {
+  
+        Restaurant restaurant = restaurantService.getRandomRestaurant();
+        int count = menuService.countMenusByRestaurant(restaurant);
```

Comment:
One crucial thing to notice, what if restaurant object is null. what will happen if null restaurant passed into `countMenusByRestaurant()`